app/controllers/users_controller.rb

```diff
@@ -1,3 +1,25 @@
 class UsersController < ApplicationController
-  def new; end
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      flash[:success] = "Welcome #{@user.email}!"
+      redirect_to '/dashboard'
+    else
+      flash[:error] = @user.errors.full_messages_to_sentence
+      if @user.errors.details.keys.include?(:email)
+        @user.email = ''
+      end
```

Comment:
Per our convo, we will delete lines 13 - 15 because it's poor UX for user's email to be deleted.